### DISCUSSION TITLE
humming AI generater

### DISCUSSION THREAD

**Discussion by tedted111** [community/community#143038] [Programming Help]
### Body

I want to make a song editor platform, but I want to put a humming function in it and make a platform that creates a beat that goes with it when you hum, unlike stable audio, when you hum a song in your head, it recommends a beat that goes with it, so I want to make an AI that makes the humming match the beat.

it gonna make something beat based of humming

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by tedted111** [comment]
please help......me......how to do...........

**Comment by DhareshP** [comment]
Creative. what tech are you using? you can do it by using librousa, pydub, tensorflow/pytorch

**Reply by tedted111** [reply]
We don't have a set technique yet, but we're really struggling to define what a good human beat is.

**Reply by DhareshP** [reply]
just start with it, you'll find the beats, capture the hum, detect the frequency using pitch detection algorithms there are libraries like pyin , tarsosdsp, these will extract pitch, to make it more aligned, use beat tracking algo's to estimate tempo, now just use ai to create beats on the trained model, use models like vae or rnn, these models learn from patterns.
hope it helps
